### ISSUE TITLE
unclear UI/UX for changing tags

### ISSUE THREAD

**Issue by tcurdt** [PDF-Archiver/PDF-Archiver#46]
**Describe the bug**

It seems like one has click the upper right "Add" again to apply changes to an already tagged file.

**To Reproduce**

1. Scan.
2. Add description and two tags.
3. Click "Add".
4. Switch to "Archive" and see the document with the two tags.
5. Switch backt to the "Tag" tab.
6. Remove a tag.
7. Switch to "Archive" and see the document still has two tags.
8. Click "Add" again.
9. Switch to "Archive" and see the document now only has one tag.

**Expected behavior**

1. Removing the tag should immediately rename the file in the archive storage.
2. The "Add" workflow from the tab is awkward. Especially as it somehow acts like queue when scanning multiple documents. Instead I would suggest to have archive have a "untagged" section where the scans end up. One then goes in tags them.
3. It would also be good to be able to swipe through the documents.

**Smartphone information:**
 - Device: iPhone 11
 - OS: 14.7
 - Version: 3.4.1

**Comment by JulianKahnert** [comment]
In "steps to reproduce":
How do you switch back to Tag in step 5? Are you using the edit button in the document view?

A sheet might be a better way, because it enforces the user to finish (e.g. save or cancel) the changes of the selected document. What do you think?

**Comment by tcurdt** [comment]
> How do you switch back to Tag in step 5? Are you using the edit button in the document view?

Na, I just used the tab bar.

> A sheet might be a better way, because it enforces the user to finish (e.g. save or cancel) the changes of the selected document. What do you think?

That doesn't really work so well with the nature of a tab bar though. And the sync nature forces an order on the user that isn't really required. I think I would rather take a take a step back to look a the flow.

In a way the "Tag" tab really is "untagged PDFs" tab.
And the "Archive" tab is the "tagged PDFs" tab.
And they are non-overlapping sets of files.
Maybe it's worth re-thinking the interfaces of these two collections.

The tricky part is making the move between the two to feel natural.
Especially if you remove tags that would turn formerly tagged files into an untagged ones.

**Comment by JulianKahnert** [comment]
You are right! The UX in this case is a little wonky. Do you have any ideas on how to improve it? I will try to step back and rethink it myself but would love to get your input as well.

A sheet overlapping the tab bar is not a consistent choice but I've found the same UX in WhatsApp: Chats (Tab) -> Create new Chat. This would be a workaround when editing tagged documents.

**Comment by tcurdt** [comment]
Happy to provide some suggestions. I'll prepare some ideas.
